<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shockley Model</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,1">
    <item>
     <layout class="QHBoxLayout" name="topButtonsLayout">
      <item>
       <widget class="QPushButton" name="loadButton">
        <property name="text">
         <string>Wczytaj CSV</string>
        </property>
       </widget>
      </item>
     <item>
       <widget class="QPushButton" name="fitButton">
        <property name="text">
         <string>Dopasuj model</string>
        </property>
       </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetZoomButton">
       <property name="text">
        <string>Resetuj zoom</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="parametersGroup">
      <property name="title">
       <string>Parametry modelu</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="labelSaturation">
         <property name="text">
          <string>Prąd nasycenia (A)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="saturationCurrentSpin">
         <property name="decimals">
          <number>12</number>
         </property>
         <property name="minimum">
          <double>1.000000000000000e-12</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000e-03</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e-09</double>
         </property>
         <property name="value">
          <double>1.000000000000000e-09</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelIdeality">
         <property name="text">
          <string>Współczynnik idealności n</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="idealityFactorSpin">
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="minimum">
          <double>5.000000000000000e-01</double>
         </property>
         <property name="maximum">
          <double>2.000000000000000e+00</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e-01</double>
         </property>
         <property name="value">
          <double>1.000000000000000e+00</double>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelTemperature">
         <property name="text">
          <string>Temperatura (K)</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="temperatureSpin">
         <property name="minimum">
          <double>2.000000000000000e+02</double>
         </property>
         <property name="maximum">
          <double>6.000000000000000e+02</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e+01</double>
         </property>
         <property name="value">
          <double>3.000000000000000e+02</double>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="labelMinVoltage">
         <property name="text">
          <string>Minimalne napięcie (V)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QDoubleSpinBox" name="minVoltageSpin">
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="minimum">
          <double>-1.000000000000000e+02</double>
         </property>
         <property name="maximum">
          <double>0.000000000000000e+00</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e-01</double>
         </property>
         <property name="value">
          <double>-5.000000000000000e+00</double>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="labelMaxVoltage">
         <property name="text">
          <string>Maksymalne napięcie (V)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QDoubleSpinBox" name="maxVoltageSpin">
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000e+00</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000e+01</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e-01</double>
         </property>
         <property name="value">
          <double>1.000000000000000e+00</double>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="labelBreakdownVoltage">
         <property name="text">
          <string>Napięcie przebicia |V_BR|</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QDoubleSpinBox" name="breakdownVoltageSpin">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000e+00</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000e+02</double>
         </property>
         <property name="singleStep">
          <double>1.000000000000000e+00</double>
         </property>
         <property name="value">
          <double>5.000000000000000e+01</double>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="labelBreakdownCurrent">
         <property name="text">
          <string>Prąd przebicia (A)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
       <widget class="QDoubleSpinBox" name="breakdownCurrentSpin">
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>-1.000000000000000e+01</double>
        </property>
        <property name="maximum">
         <double>0.000000000000000e+00</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000e-02</double>
        </property>
        <property name="value">
         <double>-1.000000000000000e-02</double>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelMethod">
        <property name="text">
         <string>Metoda</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="methodCombo">
        <item>
         <property name="text">
          <string>Analityczna (Shockley)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Metoda roznic skonczonych</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="labelSamples">
        <property name="text">
         <string>Liczba probek</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QSpinBox" name="samplePointsSpin">
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>5000</number>
        </property>
        <property name="singleStep">
         <number>50</number>
        </property>
        <property name="value">
         <number>600</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="4">
       <widget class="QCheckBox" name="showAllMethodsCheck">
        <property name="text">
         <string>Pokaz obie metody</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    </item>
    <item>
     <widget class="QChartView" name="chartView" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts/QChartView</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 </ui>
